addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/internal/utils/NetworkUtils.java

```diff
@@ -0,0 +1,304 @@
+     */
+    public LinkedHashSet<String> getNetworkIPs() {
+        return getNetworkIPs(getInterfaceIPs());
+    }
+
+    /**
+     * Takes the interfaceIPs and fetches every IP which can be assigned on their network
+     *
+     * @param networkIPs The IPs which are assigned to the Network Interfaces
+     * @return Every single IP which can be assigned on the Networks the computer is connected to
+     */
+    public LinkedHashSet<String> getNetworkIPs(Set<String> interfaceIPs) {
+        LinkedHashSet<String> networkIPs = new LinkedHashSet<String>();
+
+        for (Iterator<String> it = interfaceIPs.iterator(); it.hasNext();) {
+            try {
+                // gets every ip which can be assigned on the given network
+                SubnetUtils utils = new SubnetUtils(it.next());
+                String[] addresses = utils.getInfo().getAllAddresses();
+                for (int i = 0; i < addresses.length; i++) {
+                    networkIPs.add(addresses[i]);
+                }
+
+            } catch (Exception ex) {
+            }
+        }
+
+        return networkIPs;
+    }
+
+    /**
+     * Converts 32 bits int to IPv4 <tt>InetAddress</tt>.
+     *
+     * @param val int representation of IPv4 address
+     * @return the address object
+     */
+    public static final InetAddress int2InetAddress(int val) {
+        byte[] value = { (byte) ((val & 0xFF000000) >>> 24), (byte) ((val & 0X00FF0000) >>> 16),
+                (byte) ((val & 0x0000FF00) >>> 8), (byte) ((val & 0x000000FF)) };
+        try {
+            return InetAddress.getByAddress(value);
+        } catch (UnknownHostException e) {
+            return null;
+        }
+    }
+
+    /**
+     * Converts IPv4 <tt>InetAddress</tt> to 32 bits int.
+     *
+     * @param addr IPv4 address object
+     * @return 32 bits int
+     * @throws NullPointerException <tt>addr</tt> is <tt>null</tt>.
+     * @throws IllegalArgumentException the address is not IPv4 (Inet4Address).
+     */
+    public static final int inetAddress2Int(InetAddress addr) {
+        if (!(addr instanceof Inet4Address)) {
+            throw new IllegalArgumentException("Only IPv4 supported");
+        }
+
+        byte[] addrBytes = addr.getAddress();
+        return ((addrBytes[0] & 0xFF) << 24) | ((addrBytes[1] & 0xFF) << 16) | ((addrBytes[2] & 0xFF) << 8)
+                | ((addrBytes[3] & 0xFF));
+    }
+
+    /**
+     * Try to establish a tcp connection to the given port. Returns false if a timeout occurred
+     * or the connection was denied.
+     *
+     * @param host The IP or hostname
+     * @param port The tcp port. Must be not 0.
+     * @param timeout Timeout in ms
+     * @param logger A slf4j logger instance to log IOException
+     * @return
+     * @throws IOException
+     */
+    public boolean servicePing(String host, int port, int timeout) throws IOException {
+        try {
+            SocketAddress socketAddress = new InetSocketAddress(host, port);
```

Comment:
I assume that we close in most cases in which we open it, but a try with resource could work out well here.